baseapp/baseapp_test.go

```diff
@@ -450,8 +453,129 @@ func TestTxGasLimits(t *testing.T) {
 	header := abci.Header{AppHash: []byte("apphash")}
 
 	app.BeginBlock(abci.RequestBeginBlock{Header: header})
-	res := app.Deliver(tx)
-	assert.Equal(t, res.Code, sdk.ToABCICode(sdk.CodespaceRoot, sdk.CodeOutOfGas), "Expected transaction to run out of gas")
+	resCheck := app.Check(tx)
+	assert.Equal(t, resCheck.Code, sdk.ToABCICode(sdk.CodespaceRoot, sdk.CodeOK), "Expected abci Codetype == 0, for CodeOK")
+	resDeliver := app.Deliver(tx)
+	assert.Equal(t, resDeliver.Code, sdk.ToABCICode(sdk.CodespaceRoot, sdk.CodeOutOfGas), "Expected transaction to run out of gas due to fee in msg handler")
+	app.EndBlock(abci.RequestEndBlock{})
+	app.Commit()
+}
+
```

Comment:
So I'm a bit concerned with some of our testing and how much we rely on eg. auth/bank modules while trying to test basic functionality of the `baseapp`.

I wrote about it [here](https://github.com/cosmos/cosmos-sdk/issues/1556) and [here](https://github.com/cosmos/cosmos-sdk/issues/1555).

While it looks like you've written some great tests here, I'm afraid to include them until we've achieved the simplification we're after and can re-write these tests in a simpler way to directly test what we want about the baseapp without setting up all the keys and accounts and coins and so on.